lbplanner/services/user/get_user.php

```diff
@@ -16,78 +16,90 @@
 
 namespace local_lbplanner_services;
 
-use block_recentlyaccesseditems\external;
-use core_privacy\local\request\subsystem\plugin_provider;
+use coding_exception;
+use dml_exception;
 use external_api;
 use external_function_parameters;
 use external_single_structure;
-use external_multiple_structure;
 use external_value;
 use local_lbplanner\helpers\plan_helper;
 use local_lbplanner\helpers\user_helper;
+use core_user;
+use moodle_exception;
 
 /**
  * Get the data for a user.
+ * param userid (optional) gives back the user data with the given ID
  */
 class user_get_user extends external_api {
-    public static function get_user_parameters() {
+    public static function get_user_parameters(): external_function_parameters {
+        global $USER;
         return new external_function_parameters(array(
             'userid' => new external_value(
                 PARAM_INT,
-                'The id of the user to get the data for',
-                VALUE_REQUIRED,
-                null,
+                'The id of the user to get the data for. If not given userid is taken by Token',
```

Comment:
I think this is better wording:

'The id of the user to get the data for. If not provided it will be inferred via the token'